src/cpp/rtps/transport/UDPTransportInterface.cpp

```diff
@@ -117,43 +117,160 @@ bool UDPTransportInterface::DoInputLocatorsMatch(
     return IPLocator::getPhysicalPort(left) == IPLocator::getPhysicalPort(right);
 }
 
-bool UDPTransportInterface::init(
-        const fastrtps::rtps::PropertyPolicy*,
-        const uint32_t& max_msg_size_no_frag)
+/**
+ * @brief Set the send buffer size of the socket to the highest possible value the system allows.
+ *
+ * @param socket The socket on which to set the send buffer size.
+ * @param initial_buffer_value The initial value to try to set.
+ * @param minimum_buffer_value The minimum value to set.
+ *
+ * @return The final value set.
+ */
+static uint32_t try_setting_send_buffer_size(
```

Comment:
We can make this and its counter part template functions on `socket_base::send_buffer_size|receive_buffer_size`